tools/coverage/design.md

```diff
@@ -0,0 +1,37 @@
+# Design of Test Coverage Tool
+
+See [design.svg](https://github.com/knative/test-infra/tree/master/tools/coverage/design.svg) for the design diagram.
+
+We pack the test coverage tool in a container, that is triggered by prow. It runs test coverage profiling on target repository. Prow clones the target repository as the current working directory for the container.
+
+ Prow has handlers for different github events. We add a pre-submit prow job that is triggered by any new commit to a PR to run test coverage on the new build to compare it with the master branch and previous commit for pre-submit coverage. We add a post-submit prow job that is triggered by merge events, to run test coverage on the nodes of master branch. Test coverage data on the master branch is supplied to TestGrid for displaying the coverage change over time, as well as serve as the basis of comparison for pre-submit coverage mentioned in the pre-submit scenario.
```

Comment:
This paragraph is confusing. Since you're explaining all the scenarios below, make it a brief introduction of them instead of a summary of each. Even better, explain the main concepts so the descriptions below are less intertwined.